src/main/commandServer/httpCommandServer.ts

```diff
@@ -234,14 +261,25 @@ export class HttpCommandServer {
   }
 }
 
+type userType = 'api' | 'interactive';
+interface commandContext {
+  interactive: boolean;
+}
+
 /**
  * Description of the methods which the HttpCommandServer uses to interact with the backend.
  * There's no need to use events because the server and the core backend run in the same process.
  * The HttpCommandServer is passed an instance of this interface, and calls the methods on it
  * in order to carry out the business logic for the requests it receives.
  */
 export interface CommandWorkerInterface {
-  getSettings: () => string;
-  updateSettings: (newSettings: Record<string, any>) => Promise<[string, string]>;
-  requestShutdown: () => void;
+  getSettings: (context: commandContext) => string;
+  updateSettings: (context: commandContext, newSettings: RecursivePartial<Settings>) => Promise<[string, string]>;
+  requestShutdown: (context: commandContext) => void;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-namespace -- exporting extra things on interfaces.
```

Comment:
I assume it's a good idea to give the interface and the namespace the same name (because after typescript does its job both objects are gone). Could you add a short comment noting that this is happening, and why?